### ISSUE TITLE
HowTo have only one instance of shared modules dependencies

### ISSUE THREAD

**Issue by pdelorme** [ng-packagr/ng-packagr#810]
# Request for help

## Description
I'm trying to package a module X with dependencies to Y using ng-pacakgr.
X and Y have services and components.

App/module structure is :
```
AppModule
  - ModuleY
      - ServiceY
  - ModuleX
      - ServiceX
      - ModuleY
        - ServiceY
```
I am importing both modules X.forRoot() and Y.forRoot() in the appModule (root level).
X.forRoot() provides Y.

Packaging ModuleX with ng-packagr tells me this :

> No name was provided for external module 'Y' in options.globals – guessing 'Y'

OK.

Running the app doesn't give any error, but it seems I have two instance of ServiceY.

Looking at webpack sources in chrome, i have service Y twice :
```
- webpack://
  - ./node_modules/Y/src/y.service.js
  - X/node_modules/Y/src/y.service.js

```
positionning breakpoint in both YService's constructors show it is stopping in both classes.

**In theory Y should be instanced once in my whole application since Y is provided at the root of the app, but it seems I have two instance of services from Y.**

I tried to add the option umdModuleIds with various values but it doesn't look to make a difference.

In conclusion, I have two unique instance of YService (one being imported by app, the other imported (and redefined) by X).

Questions are :
- where is taht comming from.
- how can I be sure Y is instanciated only once in my app.
- what should I configure in umdModuleIds (how do I find moduleId)

Thanks.

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr: 2.4.2
@angular/*: 5.2.9
typescript: ~2.4.2
rxjs: 5.5.8
node: 6.10.3
npm: 3.10.10
```

**Comment by alan-agius4** [comment]
Maybe you can share a sample repo?

**Comment by pasapsi** [comment]
A similar problem.

I have 2 more different versions of the service for example
ServiceX- 1.0.0
ServiceX - 1.0.1

And the webpack refuses to collect them :((
